File: raytrace/Object/PointLight.cpp
#include "Object/PointLight.hpp"

#include "Math/Vector.hpp"
#include "Math/Transformation.hpp"

namespace Object {
	PointLight::PointLight(const Math::Point &position, const Object::Radiance &radiance)
		: mPosition(position), mRadiance(radiance)
	{
	}

	const Math::Point &PointLight::position() const
	{
		return mPosition;
	}

	const Object::Radiance &PointLight::radiance() const
	{
		return mRadiance;
	}
}

File: src/object/point_light.rs
use crate::geo;
use crate::object;

use geo::Point3;

use object::Radiance;

pub struct PointLight {
    pub position: Point3,
    pub radiance: Radiance,
}

impl PointLight {
    pub fn new(position: Point3, radiance: Radiance) -> PointLight {
        return PointLight { position, radiance };
    }
}
